components/browser/engine-gecko-beta/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt

```diff
@@ -395,17 +397,34 @@ class GeckoEngineSession(
                 (flags and GeckoSession.HistoryDelegate.VISIT_TOP_LEVEL) == 0 ||
                 (flags and GeckoSession.HistoryDelegate.VISIT_UNRECOVERABLE_ERROR) != 0) {
 
-                // Don't track visits in private mode, redirects, or error
-                // pages, even if they're top-level visits.
+                // Don't track visits in private mode, or error pages, even if they're top-level visits.
                 return GeckoResult.fromValue(false)
             }
 
+            val isReload = lastVisitedURL?.let { it == url } ?: false
+
+            val visitType = if (isReload) {
+                VisitType.RELOAD
```

Comment:
Did you ask if this is a flag/type GV could tell us about?

Reply:
Yeah; they're open to revising/expanding these flags (cc @linacambridge), but let's not block on that.